contracts/libraries/HyperdriveExecution.sol

```diff
@@ -322,6 +324,761 @@ library HyperdriveExecutionLibrary {
         flatFee = flat.mulUp(_flatFee);
     }
 
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                               Max Long                                │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    // HACK: Copied from `delvtech/hyperdrive` repo.
+    //
+    /// @dev Calculates the maximum amount of longs that can be opened.
+    /// @param _maxIterations The maximum number of iterations to use.
+    /// @return baseAmount The cost of buying the maximum amount of longs.
+    function calculateMaxLong(
+        IHyperdrive self,
+        uint256 _maxIterations
+    ) internal view returns (uint256 baseAmount) {
+        IHyperdrive.PoolConfig memory poolConfig = self.getPoolConfig();
+        IHyperdrive.PoolInfo memory poolInfo = self.getPoolInfo();
+        (baseAmount, ) = calculateMaxLong(
+            MaxTradeParams({
+                shareReserves: poolInfo.shareReserves,
+                shareAdjustment: poolInfo.shareAdjustment,
+                bondReserves: poolInfo.bondReserves,
+                longsOutstanding: poolInfo.longsOutstanding,
+                longExposure: poolInfo.longExposure,
+                timeStretch: poolConfig.timeStretch,
+                vaultSharePrice: poolInfo.vaultSharePrice,
+                initialVaultSharePrice: poolConfig.initialVaultSharePrice,
+                minimumShareReserves: poolConfig.minimumShareReserves,
+                curveFee: poolConfig.fees.curve,
+                flatFee: poolConfig.fees.flat,
+                governanceLPFee: poolConfig.fees.governanceLP
+            }),
+            self.getCheckpointExposure(latestCheckpoint(self)),
+            _maxIterations
+        );
+        return baseAmount;
+    }
+
+    // HACK: Copied from `delvtech/hyperdrive` repo.
```

Comment:
IMO this is a NOTE not a HACK. I don’t even think we need to document that this was copied.